ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -154,6 +272,8 @@ class Context:
         type_pb2.Field.TYPE_BOOL: (BoolEncoder, BoolArrayEncoder),
         type_pb2.Field.TYPE_STRING: (StringEncoder, StringArrayEncoder),
         type_pb2.Field.TYPE_ENUM: (IntEncoder, IntArrayEncoder),
+        # Array encoder is not correct, but 'None' violates expected return value
```

Comment:
Instead of using the wrong encoder, please write an encoder that returns an error.

```
def _unsupported_encoder(field_number, is_repeated, is_packed):
    raise NotImplementedError(f"Unsupported data type for field {field_number}")
```